File: java/src/main/java/mx/iteso/test/WaterIceCreamStoreTest.java
package mx.iteso.test;

import mx.iteso.Cone;
import mx.iteso.Stores.WaterIceCreamStore;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Class for water ice cream tests.
 */
public class WaterIceCreamStoreTest {
    /**
     * testing Guava Ice Cream.
     */
    @Test
    public void testGuavaWaterIceCream() {
        Cone cone = new WaterIceCreamStore().prepareCone("water", "Guava");
        assertEquals(cone.getIceCream().getFlavor(), "Guava water ice cream");
    }

    /**
     * testing Mango Ice Cream.
     */
    @Test
    public void testMangoWaterIceCream() {
        Cone cone = new WaterIceCreamStore().prepareCone("water", "Mango");
        assertEquals(cone.getIceCream().getFlavor(), "Mango water ice cream");
    }

    /**
     * testing Strawberry Ice Cream.
     */
    @Test
    public void testStrawberryWaterIceCream() {
        Cone cone = new WaterIceCreamStore().prepareCone("water", "Strawberry");
    assertEquals(cone.getIceCream().getFlavor(), "Strawberry water ice cream");
    }

    /**
     * testing Pineapple Ice Cream.
     */
    @Test
    public void testPineappleWaterIceCream() {
    Cone cone = new WaterIceCreamStore().prepareCone("water", "Pineapple");
    assertEquals(cone.getIceCream().getFlavor(), "Pineapple water ice cream");
    }
}

File: csharp/factorypattern/test/stores/WaterIceCreamStoreTest.cs
using NUnit.Framework;
using factorypattern.main;
using factorypattern.main.stores;

namespace factorypattern.test.stores
{
    internal class WaterIceCreamStoreTest
    {
        private WaterIceCreamStore _waterIceCreamStore;
        private Cone _cono;

        [SetUp]
        public void setUp()
        {
            _waterIceCreamStore = new WaterIceCreamStore();
            _cono = new Cone();
        }

        [TestCase]
        public void waterStoreTest()
        {
            _cono = _waterIceCreamStore.prepareCone("Mango");
            Assert.AreEqual("Mango Water Ice Cream", _cono.getIceCream().getName());
            _cono = _waterIceCreamStore.prepareCone("Pineapple");
            Assert.AreEqual("Pineapple Water Ice Cream", _cono.getIceCream().getName());
            _cono = _waterIceCreamStore.prepareCone("Strawberry");
            Assert.AreEqual("Strawberry Water Ice Cream", _cono.getIceCream().getName());
            _cono = _waterIceCreamStore.prepareCone("Guava");
            Assert.AreEqual("Guava Water Ice Cream", _cono.getIceCream().getName());

        }
    }
}
